contributors/design-proposals/bootstrap-discovery.md

```diff
@@ -130,6 +134,8 @@ The first part of the token is the `token-id`.  The second part is the `token-se
 
 This new type of token is different from the current CSV token authenticator that is currently part of Kubernetes.  The CSV token authenticator requires an update on disk and a restart of the API server to update/delete tokens.  As we prove out this token mechanism we may wish to deprecate and eventually remove that mechanism.
 
+The `token-id` must be 6 characters and the `token-secret` must be 16 characters.  They must be lower case ASCII letters and numbers.  Specifically it must match the regular expression: `[a-z0-9]{6}\.[a-z0-9]{16}`.  There is no strong reasoning behind this beyond the history of how this has been implemented in alpha versions.
+
 #### NEW: Bootstrap Token Secrets
 
 Bootstrap tokens are stored and managed via Kubernetes secrets in the `kube-system` namespace.  They have type `bootstrap.kubernetes.io/token`.
```

Comment:
I wanted to start as an API type from the start but after a *lot* of discussion over many months we settled on this.  You can read the super long gory history here: https://github.com/kubernetes/kubernetes/pull/30707